api-gateway-config/scripts/lua/lib/redis.lua

```diff
@@ -78,44 +82,83 @@ function _M.close(red)
   end
 end
 
+---------------------------
+----------- APIs ----------
+---------------------------
+
+--- Add API to redis
+-- @param red Redis client instance
+-- @param id id of API
+-- @param apiObj the api to add
+function _M.addAPI(red, id, apiObj)
+  local ok, err = red:hset("apis", id, apiObj)
+  if not ok then
+    request.err(500, utils.concatStrings({"Failed adding API to redis: ", err}))
```

Comment:
Probably best to return a generic error here like `Failed to save the API`, while still logging the actual reason. Same comment for the other pieces of this API: `Failed to retrieve APIs`, etc.